Add plugin: Jira Issue Manager

Alamion opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/Alamion/obsidian-jira-sync

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
What sets your plugin apart from the other Jira plugins?
Please include this information in your README.

[Copyright (c) 2024 aperez](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/LICENSE#L3)
Update this.

[export function htmlToMarkdown(str: any): string {](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/tools/markdown_html.ts#L5-L15)
The Obsidian API exports a function with the same name, use that instead of importing a third-party library.

[await plugin.app.vault.process(file, (fileContent) => {](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/tools/mappingObsidianJiraFields.ts#L155-L195)
Use the `processFrontMatter` function for this instead of manually parsing YAML.

[containerEl.createEl("h3", { text: "Jira-Obsidian field mapping" });](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/settings/JiraSettingTab.ts#L150)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

[this.contentEl.createEl("h2", { text: "Add Work Log" });](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/modals/issueWorkLogModal.ts#L69), [.setName("Time Spent")](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/modals/issueWorkLogModal.ts#L73), [.setName("Start Time")](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/modals/issueWorkLogModal.ts#L85), [.setName("Work Description")](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/modals/issueWorkLogModal.ts#L162), [this.contentEl.createEl("h2", {text: "Search Issue by Key"});](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/modals/IssueSearchModal.ts#L16), [.setName("Issue Key")](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/modals/IssueSearchModal.ts#L19)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[callback: async () => {](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/commands/getIssue.ts#L28), [callback: async () => {](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/commands/createIssue.ts#L14), [callback: async () => {](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/commands/updateIssue.ts#L13), [callback: async () => {](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/commands/updateStatus.ts#L16), [callback: async () => {](https://github.com/Alamion/obsidian-jira-sync/blob/60e3a40bdbb5f5a7bcc05f6857cf5830e0989ade/src/commands/addWorkLog.ts#L16)
These should be using [conditional commands](https://docs.obsidian.md/Plugins/User+interface/Commands#Conditional+commands)

Alamion commented:
Hello, @joethei
Thanks a lot for your review. I fixed all the problems you mentioned. However, I want to make sure everything's right with README. I reworked it's structure from scratch, hope now it became more clear why I need seventh Jira-related plugin in Obsidian store. I'll be glad to hear constructive criticism if there's anything left unnoticed.

joethei commented:
[# Obsidian Jira Plugin](https://github.com/Alamion/obsidian-jira-sync/blob/32273539706907a32fc2cbcad68ef6a4dfd9d37b/README.md?plain=1#L1)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

Alamion commented:
Done
